noxfile.py

```diff
@@ -180,3 +180,27 @@ def make(session: nox.Session):
         *(args.make_args or ("clean", "coredocs")),
     ]
     session.run("make", "-C", "docs/docsite", *make_args, external=True)
+
+
+@nox.session
+def tag(session: nox.Session):
+    """
+    Determine the missinlg ansible-core releases,
+    create corresponding tags for each release in the ansible-documentation repo,
+    and then push them
+    """
+    install(session, req="tag")
+
+    DEFAULT_REPO = "https://github.com/ansible/ansible"
+
+    args = list(session.posargs)
+    if not any(arg.startswith("--core") for arg in args):
+        tmpdir = session.create_tmp()
+        session.run("git", "clone", "--quiet", DEFAULT_REPO, tmpdir, external=True)
+        args.extend(["--core", tmpdir])
```

Comment:
I'm not sure about cloning the core repo as part of this session. The tagger script already determines the [default core checkout](https://github.com/ansible/ansible-documentation/blob/aab570123955229d8bc2fd66e7a73c5812f02044/hacking/tagger/tag.py#L42) and an extra clone feels wasteful.

This also requires you to set the `--core` argument when running the tagger if you want to use the existing core checkout, which seems a bit less convenient than using the tagger directly.

Maybe we could call `tag.py` to catch when the `ansible` repo doesn't exist and then do the clone?

Reply:
Hmm. Now I'm on the fence. I guess the clone here resolves the need to set the `--remote` arg if it's not `origin` as well as making sure that the repo is up to date.

Reply:
Yeah, I had a lot of the same thoughts. 

What I landed on for a first pass was designing around the idea of 

"running `nox -s tag` should require little/no knowledge of how the tagging tool works under the hood, and should 'just work'"

Based on running the tag tool multiple times, I'm fairly confident I know (relative to the docs directory) where it expects core to be. I can add in a test for if it already exists either there or in the tmp location, and potentially some other checks to confirm branches etc.

Reply:
@x1101 that makes sense and maybe I'm coming around to the idea of doing the clone here. One thing I think it's important to avoid is making the nox session too complex and my suggestions could be pushing things down the wrong path. The nox session should only be a light wrapper to the tagger script.